### PR TITLE
Update pkg/api/testapi because namespace is always in the path now

### PR DESCRIPTION
**Pull request by caesarxuchao**
#8087

@nikhiljindal  @bgrant0607 @krousey

### CODE REVIEW

**Comment by bgrant0607** [comment]
cc @smarterclayton

**Comment by erictune** [comment]
@lavalamp @smarterclayton who wants this PR?

**Comment by nikhiljindal** [comment]
I can take this.

File: pkg/api/testapi/testapi.go [hunk 1]
```diff
@@ -86,13 +85,11 @@ func ResourcePathWithPrefix(prefix, resource, namespace, name string) string {
...
-	if !api.PreV1Beta3(Version()) {
-		if namespace != "" {
-			path = path + "/namespaces/" + namespace
-		}
-		// Resource names in v1beta3 are lower case.
-		resource = strings.ToLower(resource)
+	if namespace != "" {
+		path = path + "/namespaces/" + namespace
 	}
+	// Resource names in v1beta3 are lower case.
```
Comment (nikhiljindal) [inline pkg/api/testapi/testapi.go]: Can change this to "Resource names are lower case"

File: pkg/apiserver/handlers_test.go [hunk 2]
```diff
@@ -54,7 +54,7 @@ func getPath(resource, namespace, name string) string {
 }
 
 func pathWithNamespaceQuery(resource, namespace, name string) string {
```
Comment (nikhiljindal) [inline pkg/apiserver/handlers_test.go]: Replace usage of this function with getPath and delete this function

File: pkg/apiserver/handlers_test.go [hunk 3]
```diff
@@ -62,7 +62,7 @@ func pathWithPrefix(prefix, resource, namespace, name string) string {
 }
 
 func pathWithPrefixAndNamespaceQuery(prefix, resource, namespace, name string) string {
```
Comment (nikhiljindal) [inline pkg/apiserver/handlers_test.go]: Replace usage of this function with pathWithPrefix and delete this function

File: test/integration/auth_test.go [hunk 4]
```diff
@@ -79,7 +79,7 @@ func path(resource, namespace, name string) string {
 }
 
 func pathWithNamespaceQuery(resource, namespace, name string) string {
```
Comment (nikhiljindal) [inline test/integration/auth_test.go]: Replace this function with path() and delete this.

File: test/integration/auth_test.go [hunk 5]
```diff
@@ -91,7 +91,7 @@ func timeoutPath(resource, namespace, name string) string {
 }
 
 func timeoutPathWithNamespaceQuery(resource, namespace, name string) string {
```
Comment (nikhiljindal) [inline test/integration/auth_test.go]: Replace this function with timeoutPath()

**Comment by nikhiljindal** [comment]
LGTM, module comments

**Comment by caesarxuchao** [comment]
Thank you @nikhiljindal! Updated, PTAL.

**Comment by nikhiljindal** [comment]
LGTM, thanks!

**Pull request merged**